LWCOSC/force-app/main/default/classes/ProductServiceImpl.cls

```diff
@@ -0,0 +1,97 @@
+public class ProductServiceImpl implements IProductService {
+    
+    public List<Purchase_Order__c> getPurchaseOrders(String userId){
+        List<Purchase_Order__c> poList = new List<Purchase_Order__c>();
+        PurchaseOrderSelectorImpl poSelectorImpl = new PurchaseOrderSelectorImpl();
+        try{ 
+            poList = poSelectorImpl.getPurchaseOrdersByUserId(userId);
+            return  poList;
+        }
+        catch(Exception ex){
+            System.System.debug('Exception in ProductServiceImpl in method getPurchaseOrders= '+ ex.getMessage());
+            return poList;
+        }
+    }
+
+    public List<Product2> getProducts(String searchKey, String sortBy, String sortDirection){
+        List<Product2> prodList = new List<Product2>();
+        ProductSelectorImpl prodSelImpl = new ProductSelectorImpl();
+        try{
+             prodList = prodSelImpl.getProducts(searchKey, sortBy,  sortDirection);
+             return prodList;
+        }
+        catch(Exception ex){
+            System.System.debug('Exception in ProductServiceImpl at Method getProducts ' + ex.getMessage());
+            return prodList;
+        }
+       
+    }
+
+    public List<String> updatePurchaseOrderRecords(List<map<String, Object>> objList, Id userId){
+        List<String> result = new List<String>();
+        ProductSelectorImpl prodSelImpl = new ProductSelectorImpl();
+        fflib_SObjectUnitOfWork uow = new fflib_SObjectUnitOfWork(
+            new Schema.SObjectType[] {
+                Product2.SObjectType,
+                Purchase_Order__c.SObjectType,
+                Purchase_Order_Line_Item__c.SObjectType
+            }
+        );
+        Map<Id, Decimal> prodMap = new  Map<Id, Decimal>();
+
+        for(Integer i=0; i<objList.size();i++){
+            ID id = (Id)objList[i].get('Id');
+            Decimal unit = (Decimal)objList[i].get('Units');
+            prodMap.put(id, unit);
+        }
+        
+        //List<Product2> prodIdList = prodSelImpl.selectById(prodMap.keySet());
+        List<Product2> prodIdList = [SELECT Id, Available_Quantity__c 
```

Comment:
should use selector instead of querying directly

Reply:
Point Noted, As discussion No action require